### DISCUSSION TITLE
Improve Search functionality

### DISCUSSION THREAD

**Discussion by PastorHope7** [damionx7/Zettel-Notes-Documentation#225] [Feature Requests]
The search function is the backbone of managing huge files. 
**Zettelk Notes** search function would be better if it could search multiple terms at a time.

Examples:
- Google keep 
- Unotes with tags (These apps have an idea of a good search function) 

If multiple tags could be search, it'll be a more useful search feature.

**Comment by damionx7** [comment]
https://github.com/damionx7/Zettel-Notes-Documentation/discussions/76

**Comment by damionx7** [comment]
https://github.com/damionx7/Zettel-Notes-Documentation/issues/357

**Comment by nathanl** [comment]
If I search my notes using `git grep`, I can see occurrences of a term as they appear in a file. [Obsidian](https://obsidian.md/) does this too, and lets me click to open a specific file, even at the specific scroll position of the match I clicked on:

![Image](https://github.com/user-attachments/assets/e818ec47-265e-4c8e-831e-db688c982d47)

When I search in Zettel, I see only the names of files in which the term appears, which doesn't give me much context. Pressing enter just opens the first of those.

I'd love for the search feature to work the way Obsidian's does.
